contest/remote/vmksft-p.py

```diff
@@ -97,12 +98,23 @@ def _parse_nested_tests(full_run):
             continue
 
         v = result_re.match(line).groups()
+        time_ms = 0
         name = v[3]
-        if len(v) > 5 and v[5]:
+        if len(v) > 5 and v[4] and v[5]:
             if v[5].lower().startswith('skip') and result == "pass":
```

Comment:
According to the specs, to mark a test as skip, the line should start with `ok` (only in TAP, even if it is not really clear, not forced in KTAP → maybe we could remove the `result == "pass" then`), and `skip` should be the first word after the `#`:

```
Test case result lines indicate the final status of a test. They are required and must have the format:

    <result> <number> [<description>][ # [<directive>] [<diagnostic data>]]

(...)

Currently accepted directives are:

“SKIP”, which indicates a test was skipped (note the result of the test case result line can be either “ok” or “not ok” if the SKIP directive is used)
```

https://docs.kernel.org/dev-tools/ktap.html#test-case-result-lines
https://testanything.org/tap-version-13-specification.html


So I think we should keep the `startswith('skip')`, but we can drop the `result == "pass"`. Do you want me to do that?

Reply:
> Are you assuming `time=x` is not a directive?

Correct, it is not, and it should not be I think, because it can be used with the other ones (ERROR, TIMEOUT, XFAIL), and it doesn't have a fixed value.

> I thought that `diagnostic data` is for random debug messages like "toolX is not installed".

Correct, anything can be added there.

> Can we make `time=x` part of the spec, then

We can try I suppose, but it looks like nothing has been defined there to let people the freedom to do anything they want. But in our case here, we would prefer to suggest a format, so people do the same, and we can extract such info.

Maybe we can add it as an example on how to use it?, similar to their last example:

```
    ok 5 check return code # rcode=0

The test “check return code” passed, with additional diagnostic data “rcode=0”
```

> or it will be a free form string we happen to use in networking? Link KSFT_SLOW_MACHINE etc?

Or we start by using `time=Xms` in different selftests, and then we sugggest to modify the spec to adopt this format to display the time it took per test? :)